src/file_parsing.jl

```diff
@@ -0,0 +1,327 @@
+using TOML
+using DocStringExtensions
+
+export ParamDict
+export parse_toml_file,
+    get_parametric_type,
+    iterate_alias,
+    log_component!,
+    get_values,
+    get_parameter_values!,
+    get_parameter_values,
+    check_override_parameter_usage,
+    write_log_file,
+    log_parameter_information,
+    merge_override_default_values,
+    create_parameter_struct
+        
+    
+"""
+    ParamDict{FT}
+
+structure to hold information read-in from TOML file, as well as a parametrization type `FT`
+# Constructors
+
+    ParamDict(data::Dict, dict_type::String, override_dict::Union{Nothing,Dict})
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+
+"""
+struct ParamDict{FT}
+    "dictionary representing a default/merged parameter TOML file"
+    data::Dict
+    "string to determine how dictionary look-up is performed"
+    dict_type::String
```

Comment:
We should instead make this a type parameter, or just have two different structs and dispatch on them. cc @odunbar